.github/actions/run-ios-simulator/src/main.ts

```diff
@@ -12,61 +12,62 @@ async function run(): Promise<void> {
         const iphoneToSimulate = core.getInput("iphoneToSimulate", { required: false });
         const args = core.getInput("arguments", { required: false });
 
-        let runtimeId = "";
-        const options : exec.ExecOptions = {};
-        options.listeners = {
-            stdout: (data: Buffer) => {
-                runtimeId += data.toString();
-            },
-        };
-
-        await execCmd("xcrun simctl list runtimes", options);
+        let runtimeId = await execCmd("xcrun simctl list runtimes");
 
         // Sample output: iOS 14.5 (14.5 - 18E182) - com.apple.CoreSimulator.SimRuntime.iOS-14-5
-        // and we want to extract "iOS 14.5"
+        // and we want to extract "iOS 14.5" and "com.apple.CoreSimulator.SimRuntime.iOS-14-5"
         // If we want to allow launching watchOS/tvOS simulators, replace the 'iOS' with an 'os' argument
-        const matches = runtimeId.match(/(iOS \d{1,2}(.\d{1,2})?)/g);
-        if (matches && matches.length > 0) {
-            runtimeId = matches[0].replace(" ", "");
-            core.info(`runtimeId: ${runtimeId}`);
+        const matches =/(?<runtime1>iOS \d{1,2}(.\d{1,2})?).*(?<runtime2>com\.apple\.CoreSimulator\.SimRuntime\.iOS-[0-9.-]+)/g.exec(runtimeId);
+        if (!matches || matches?.length == 0 || matches?.groups || matches?.groups?.runtime1.length == 0 || matches?.groups?.runtime2.length == 0) {
+            core.setFailed(`Impossible to fetch a runtime. Check runtimes and retry.\n${runtimeId}`);
+            return;
         }
+        core.info(`runtimeId: ${runtimeId}`);
 
         try {
-            await execCmd(`xcrun simctl create ${id} com.apple.CoreSimulator.SimDeviceType.${iphoneToSimulate} ${runtimeId.toString()}`);
+            runtimeId = matches?.groups?.runtime1?.replace(" ", "") ?? "";
```

Comment:
Those question marks are not necessary since we already checked that the value of `runtime1` is truthy.

Reply:
Unfortunately, it was a result of the mistake done in the nullity check. Rettified now.